envoy/stream_info/filter_state.h

```diff
@@ -75,6 +75,11 @@ class FilterState {
     // upstream connections. Note that this affects connection pooling,
     // preventing any re-use of the upstream connections in the worst case.
     SharedWithUpstreamConnection,
+
+    // Same as SharedWithUpstreamConnection, except that the filter state is
+    // not transitively shared. The filter state is imported into the upstream
+    // connection filter state as exclusive to the upstream connection.
+    SharedWithUpstreamConnectionOnce,
```

Comment:
So, this means with loop back listeners as 1 (decap) -> 2 -> 3 (encap),  the filterstate sharing is only available to 2?

Reply:
Another question is that if the config only has one loop back listeners like 1(decap) -> 2(encap), then this change is no-op, right?

Reply:
Yes to both questions.